lib/connectors/memory.js

```diff
@@ -54,10 +55,38 @@ Memory.prototype.getTypes = function() {
 };
 
 Memory.prototype.connect = function(callback) {
+  var self = this;
+  if (this.connected) {
+    // It's already connected or being connected
+    // Do NOT run loadFromFile again as it will override the cache
+    if (callback) process.nextTick(callback);
+    return;
+  }
+  if (callback) {
+    // Queue the callback
+    this.pendingConnectCallbacks = this.pendingConnectCallbacks || [];
```

Comment:
Could you please elaborate a little bit on the reasons for repeating `pendingConnectCallbacks` implementation in both the memory connector and datasource? I also noticed that the implementation is slightly different in each place, is that intentional?